### PR TITLE
Switch to using Visual Studio's version of LLVM and Clang

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Description

This pull request includes changes to the `.github/workflows/reusable-build.yml` file to streamline the build process by removing unnecessary steps and adding new ones for setting up Visual Studio LLVM.

Build process improvements:

* Removed setup for the choco cache folder and caching of choco packages. (`.github/workflows/reusable-build.yml`)
* Added a step to add Visual Studio LLVM to the PATH. (`.github/workflows/reusable-build.yml`)
* Replaced the installation of LLVM tools with a check for the Clang version. (`.github/workflows/reusable-build.yml`)

## Testing

CI/CD

## Documentation

TBD

## Installation

No.

### CODE REVIEW

File: .github/workflows/reusable-build.yml [hunk 1]
```diff
@@ -133,32 +133,15 @@ jobs:
...
 
-    - name: Setup choco cache folder
-      # Set the choco cache to a local folder so that it can be cached.
+    - name: Add Visual Studio LLVM to path
       if: steps.skip_check.outputs.should_skip != 'true'
-      id: choco-cache
       run: |
-        mkdir ${{github.workspace}}\choco_cache
-        choco config set --name cacheLocation --value ${{github.workspace}}\choco_cache
+        echo "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\llvm\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
```
Comment (dthaler) [inline .github/workflows/reusable-build.yml:139]: I don't think we should hard code path names in the middle of job commands, use variables instead (yes i see line 160 already did previously so it's not new in this PR).

File: docs/GettingStarted.md [hunk 2]
```diff
@@ -234,7 +233,7 @@ On a defender machine with [eBPF installed](#installing-ebpf-for-windows), do th
 1. Install and set up a DNS server.
 1. Make sure that either test signing was enabled as discussed in
    [Installing eBPF for Windows](#installing-ebpf-for-windows), or the kernel debugger (KD) is attached and running.
-1. Install [clang](https://github.com/llvm/llvm-project/releases/download/llvmorg-11.0.1/LLVM-11.0.1-win64.exe)
+1. Install [clang](https://github.com/llvm/llvm-project/releases/download/llvmorg-17.0.3/LLVM-17.0.3-win64.exe)
```
Comment (dthaler) [inline docs/GettingStarted.md:236]: Does LLVM 17 work with cilium?  I thought it didn't.

File: ebpf-for-windows.sln [hunk 3]
```diff
@@ -217,11 +219,14 @@ Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "ebpf_store_helper", "libs\s
 EndProject
 Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "usersim", "external\usersim\src\usersim.vcxproj", "{030A7AC6-14DC-45CF-AF34-891057AB1402}"
 EndProject
-Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "libbtf", "external\ebpf-verifier\build\external\libbtf\libbtf\libbtf.vcxproj", "{0F5503EA-1626-3475-AE16-C5E945187878}"
+Project("{8BC9CEB8-8B4A-11D0-8D11-00A0C91BC942}") = "libbtf", "external\ebpf-verifier\build\external\libbtf\libbtf\libbtf.vcxproj", "{9FD0723C-6979-3D32-8D17-6CECA34B7026}"
```
Comment (dthaler) [inline ebpf-for-windows.sln:222]: Why did the GUID change?

File: docs/GettingStarted.md (hunk 2 above)
Comment (Alan-Jowett) [reply docs/GettingStarted.md:236]: Unfortunately, supply chain security trumps this. Folks consuming the Cilium BPF programs we will either need to fix prevail or the BPF programs.

We cannot deploy a production version of our products based on a code base that isn't getting security fixes. No one is signed up to provide security fixes for LLVM 11.0.1, but Microsoft's Visual Studio team is providing security fixes for LLVM 17.0.3 (at least the version that ships with Visual Studio).

File: ebpf-for-windows.sln (hunk 3 above)
Comment (Alan-Jowett) [reply ebpf-for-windows.sln]: Reverted.

File: .github/workflows/reusable-build.yml (hunk 1 above)
Comment (Alan-Jowett) [reply .github/workflows/reusable-build.yml]: Switch to derive clang.exe path from cl.exe path.

File: docs/GettingStarted.md (hunk 2 above)
Comment (dthaler) [reply docs/GettingStarted.md:236]: Sounds like a good reason to me.

File: scripts/Setup-DevEnv.ps1 [hunk 4]
```diff
@@ -8,11 +8,11 @@ choco install git --version 2.38.1 -y
 choco install visualstudio2022community --version 117.4.2.0 -y
 choco install visualstudio2022-workload-nativedesktop --version 1.0.0 -y
 choco install visualstudio2022buildtools --version 117.4.2.0 -y
+vs_installer.exe modify --installpath "C:\Program Files\Microsoft Visual Studio\2022\Community" --add Microsoft.VisualStudio.Component.VC.Llvm.ClangToolset --quiet
```
Comment (dthaler) [inline scripts/Setup-DevEnv.ps1:11]: This should not assume that `%ProgramFiles%` is on the C drive.

Comment (Alan-Jowett) [reply scripts/Setup-DevEnv.ps1:11]: Good catch. Let me see if I can apply the same fix as I did to the yaml.

**Review by dthaler — DISMISSED** [review]

File: docs/GettingStarted.md (hunk 2 above)
Comment (Alan-Jowett) [reply docs/GettingStarted.md:236]: With a few very minor bug fixes in the Cilium code, it now compiles using LLVM 17!

**Review by dthaler — APPROVED** [review]

**Review by shpalani — DISMISSED** [review]

**Comment by Alan-Jowett** [comment]
@shpalani and @dthaler please re-approve after I had to resolve a merge conflict from main.

**Review by shpalani — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Pull request merged**